pkg/planner/cascades/task/task_opt_group_expression.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package task
+
+import (
+	"io"
+
+	"github.com/pingcap/tidb/pkg/planner/cascades/base"
+	"github.com/pingcap/tidb/pkg/planner/cascades/memo"
+	"github.com/pingcap/tidb/pkg/planner/cascades/rule"
+	"github.com/pingcap/tidb/pkg/planner/pattern"
+)
+
+var _ base.Task = &OptGroupTask{}
+
+type OptGroupExpressionTask struct {
+	BaseTask
+
+	groupExpression *memo.GroupExpression
+	// currently for each opt expression, it should be explore-type.
+}
+
+// NewOptGroupExpressionTask return a targeting optimizing group expression task.
+func NewOptGroupExpressionTask(mctx *memo.MemoContext, ge *memo.GroupExpression) *OptGroupExpressionTask {
+	return &OptGroupExpressionTask{
+		BaseTask:        BaseTask{mctx: mctx},
+		groupExpression: ge,
+	}
+}
```

Comment:
I guess that this object's lifetime is short and may have great impacts on GC?

Reply:
Do you mean the different-type tasks, make sense, currently we only reuse the stack container, maybe the element inside which is as you mentioned here, we should manage a pool of reusing Task obj.

do you think we should manage this Task pool across different sessions (global), or just limited in a session which is only save the reallocation for the same type task which may be just popped before from the stack.